library/Icinga/Web/Controller/ActionController.php

```diff
@@ -450,6 +451,22 @@ public function preDispatch()
             $form->handleRequest();
         }
         $this->_helper->layout()->autoRefreshForm = $form;
+
+        if (! $this->isXhr()
+            // Layout is only enabled for initial page loads or page reloads
+            && $this->_helper->layout()->isEnabled()
+            && Config::app()->get('security', 'use_strict_csp', false)
+        ) {
+            $response = $this->getResponse();
+            // Window ID must be explicitly set here for the window-aware CSP nonces to work,
+            // since we only set the window ID upon XHR requests.
+            if (! $response->getHeader('X-Icinga-WindowId')) {
+                $response->setHeader('X-Icinga-WindowId', $this->Window()->getId(), true);
```

Comment:
I don't quite understand why this header is needed. It won't be processed anyway, if it's really a page load. Can you explain when/where the header is required?